### PR TITLE
Bypass checks for specifique author

### PR DESCRIPTION
**Pull request by erwan-b**

### CODE REVIEW

File: bert_e/settings.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ class SettingsSchema(Schema):
     need_author_approval = fields.Bool(missing=True)
     required_leader_approvals = fields.Int(missing=0)
     required_peer_approvals = fields.Int(missing=2)
+    pr_options = fields.Dict(missing={})
```
Comment (tcarmet) [inline bert_e/settings.py:103]: I would rename this config with something in regards to the `author` as this option is not only about the PR but mostly about the author of a PR.

**Review by tcarmet — COMMENTED** [review]
This is looking good, here's my advice to move on:

I see a lot of comment regarding " add the option here" "add the option there"

I would advise for you to look if there's a way you can set the value in `job.settings.myoption` like `job.settings.bypass_author_approval`

That way you don't have to modify any of the condition, you just need to make sure those variable got the value.

Check how options are set and see if that helps.

**Comment by tcarmet** [comment]
Please make sure that those variables are available in `active_options` so that when a PR comment appears the active option appears to the user as well.

File: bert_e/tests/test_bert_e.py [hunk 2]
```diff
@@ -2472,6 +2472,259 @@ def test_bypass_options(self):
                 options=self.bypass_all_but(['bypass_incompatible_branch']),
                 backtrace=True)
 
+    def test_bypass_user_options(self):
```
Comment (tcarmet) [inline bert_e/tests/test_bert_e.py:2475]: Rename the tests name with something that talks about the `pr_author_options`

**Review by cmeron-scality — APPROVED** [review]

File: bert_e/tests/test_bert_e.py [hunk 3]
```diff
@@ -2472,6 +2472,413 @@ def test_bypass_options(self):
...
+    - bypass_author_approvall
+    - bypass_build_status
+    - bypass_peer_approval
+"""  # noqa
+        pr = self.create_pr('feature/TEST-00014', 'development/4.3')
+        with self.assertRaises(exns.IncorrectSettingsFile):
+            self.handle(
+                pr.id,
+                settings=settings,
+                backtrace=True)
```
Comment (tcarmet) [inline bert_e/tests/test_bert_e.py:2498]: perfect

File: bert_e/tests/test_bert_e.py [hunk 4]
```diff
@@ -2472,6 +2472,413 @@ def test_bypass_options(self):
...
+required_leader_approvals: 1
+required_peer_approvals: 2
+project_leaders:
+  - {admin}
+pr_author_options:
+  {contributor}:
+    - bypass_build_status
+    - bypass_peer_approval
+    - bypass_jira_check
+"""  # noqa
```
Comment (tcarmet) [inline bert_e/tests/test_bert_e.py:2863]: I don't see any jira config inside this settings to ensure that bert-e is checking jira. Is it set elsewhere? On top of that we're not check if the `self.handle` is a success or not.

**Review by tcarmet — COMMENTED** [review]
Almost done.

Need to address one comment I made.
And lastly please update the Changelog file with the version that is going to have your feature released and small line explaining it.

Good job though! :tada: we'll be able to merge this tomorrow!

File: CHANGELOG [hunk 5]
```diff
@@ -1,6 +1,10 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## [3.5.0] - 2021-04-27
+# Added
+- Add "pr_author_options" option to config, witch add bypass to a specifique user
```
Comment (tcarmet) [inline CHANGELOG:6]:
```suggestion
- Add "pr_author_options" option to config, which add bypass to a specifique user
```

**Review by tcarmet — APPROVED** [review]

**Pull request merged**